app/models/participant_declaration.rb

```diff
@@ -21,6 +21,7 @@ class ParticipantDeclaration < ApplicationRecord
   scope :npq, -> { joins(:current_profile_declaration).merge(ProfileDeclaration.npq_profiles) }
   scope :payable, -> { joins(:current_profile_declaration).merge(ProfileDeclaration.where(payable: true)) }
   scope :unique_id, -> { select(:user_id).distinct }
+  scope :for_participant, ->(*args) { where(user_id: args[0]) if args.any? }
```

Comment:
I don't think this is the place to check that args exist. I think that should be done in the controller, and this scope should require a user id.

Reply:
ok, made it similar to npq application scope